WooCommerce/Classes/POS/Presentation/Reusable Views/POSModalManager.swift

```diff
@@ -1,15 +1,26 @@
 import SwiftUI
+import Combine
 
 class POSModalManager: ObservableObject {
     @Published private(set) var isPresented: Bool = false
     @Published private(set) var allowsInteractiveDismissal: Bool = true
     private var contentBuilder: (() -> AnyView)?
     private var onDismiss: (() -> Void)?
+    private var subscriptions = Set<AnyCancellable>()
 
     func present<Content: View>(onDismiss: @escaping () -> Void, content: @escaping () -> Content) {
         contentBuilder = { AnyView(content()) }
```

Comment:
I have seen no difference on my testing, but perhaps are we holding a reference to the content closure here? 
```
        contentBuilder = { [weak self] in
            guard let self else { return AnyView(EmptyView()) }
            return AnyView(content())
        }
```

Reply:
> perhaps are we holding a reference to the content closure here

Yes, I think so. But I don't know how to break the reference cycle, since, as I understand, it's not with closure itself but what the closure holds deeper within the hierarchy.

So POSModalManager ends up holding POSModalManager

![image](https://github.com/user-attachments/assets/663f5e84-20f2-4ac9-8a61-b572f43c60f8)

Reply:
`[weak self]` doesn't make a difference here, since we're not capturing `self` within a closure.